### PR TITLE
Apply EIP-5656, MCOPY

### PR DESCRIPTION
**Pull request by sjnam**
## Proposed changes

This PR applies [EIP-5656](https://github.com/ledgerwatch/erigon/pull/7887), mcopy instruction

## Types of changes

- [ ] Bugfix
- [x] New feature or enhancement
- [ ] Others

## Checklist

- [x] I have read the [CONTRIBUTING GUIDELINES](https://github.com/klaytn/klaytn/blob/master/CONTRIBUTING.md) doc
- [x] I have signed the [CLA](https://cla-assistant.io/klaytn/klaytn)
- [x] Lint and unit tests pass locally with my changes (`$ make test`)
- [x] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Comment by hyunsooda** [comment]
Could you please update the function `NewEVMInterpreter()` as well?

File: blockchain/vm/opcodes.go [hunk 1]
```diff
@@ -123,6 +123,7 @@ const (
 	MSIZE
 	GAS
 	JUMPDEST
+	MCOPY
```
Comment (ian0371) [inline blockchain/vm/opcodes.go:126]: Isn't this 0x5c? I think we need 0x5e

File: blockchain/vm/eips.go [hunk 2]
```diff
@@ -217,3 +219,29 @@ func enable3860(jt *JumpTable) {
 	jt[CREATE].dynamicGas = gasCreateEip3860
 	jt[CREATE2].dynamicGas = gasCreate2Eip3860
 }
+
+// enable5656 enables EIP-5656 (MCOPY opcode)
+// https://eips.ethereum.org/EIPS/eip-5656
+func enable5656(jt *JumpTable) {
+	jt[MCOPY] = &operation{
```
[the remark this answers is not in the document]
Comment (blukat29) [reply blockchain/vm/eips.go:226]: How about using the same computation cost as CALLDATACOPY (100)?
Both do pretty much the same thing, i.e. `copy(memory, ...)`

File: blockchain/vm/opcodes.go (hunk 1 above)
Comment (blukat29) [reply blockchain/vm/opcodes.go:126]: We could add filler enums until we have TLOAD (0x5C) and TSTORE (0x5D)
```suggestion
	OP_5C
	OP_5D
	MCOPY
```

File: blockchain/vm/eips.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (sjnam) [reply blockchain/vm/eips.go:226]: @yoomee1313 @blukat29 
I executed the command `go test -bench=BenchmarkOpMcopy -benchtime 200x` 10 times according to
[benchmark test](https://klaytn-foundation.atlassian.net/wiki/spaces/KNG/pages/8820249/gx+2021.09.09+computation+cost#2.-%EB%B2%A4%EC%B9%98%EB%A7%88%ED%81%AC-%EA%B8%B0%EB%B0%98-%ED%85%8C%EC%8A%A4%ED%8A%B8).
```
go test -bench=BenchmarkOpMcopy -benchtime 200x
goos: darwin
goarch: arm64
pkg: github.com/klaytn/klaytn/blockchain/vm
BenchmarkOpMcopy-10    	     200	       159.2 ns/op
PASS
ok  	github.com/klaytn/klaytn/blockchain/vm	0.425s
--
goos: darwin
goarch: arm64
pkg: github.com/klaytn/klaytn/blockchain/vm
BenchmarkOpMcopy-10    	     200	       137.3 ns/op
PASS
ok  	github.com/klaytn/klaytn/blockchain/vm	0.511s
--
goos: darwin
goarch: arm64
pkg: github.com/klaytn/klaytn/blockchain/vm
BenchmarkOpMcopy-10    	     200	       117.1 ns/op
PASS
ok  	github.com/klaytn/klaytn/blockchain/vm	0.425s
--
goos: darwin
goarch: arm64
pkg: github.com/klaytn/klaytn/blockchain/vm
BenchmarkOpMcopy-10    	     200	       132.1 ns/op
PASS
ok  	github.com/klaytn/klaytn/blockchain/vm	0.429s
--
goos: darwin
goarch: arm64
pkg: github.com/klaytn/klaytn/blockchain/vm
BenchmarkOpMcopy-10    	     200	       152.1 ns/op
PASS
ok  	github.com/klaytn/klaytn/blockchain/vm	0.424s
--
goos: darwin
goarch: arm64
pkg: github.com/klaytn/klaytn/blockchain/vm
BenchmarkOpMcopy-10    	     200	       154.8 ns/op
PASS
ok  	github.com/klaytn/klaytn/blockchain/vm	0.429s
--
goos: darwin
goarch: arm64
pkg: github.com/klaytn/klaytn/blockchain/vm
BenchmarkOpMcopy-10    	     200	       121.0 ns/op
PASS
ok  	github.com/klaytn/klaytn/blockchain/vm	0.427s
--
goos: darwin
goarch: arm64
pkg: github.com/klaytn/klaytn/blockchain/vm
BenchmarkOpMcopy-10    	     200	       149.8 ns/op
PASS
ok  	github.com/klaytn/klaytn/blockchain/vm	0.425s
--
goos: darwin
goarch: arm64
pkg: github.com/klaytn/klaytn/blockchain/vm
BenchmarkOpMcopy-10    	     200	       139.4 ns/op
PASS
ok  	github.com/klaytn/klaytn/blockchain/vm	0.422s
--
goos: darwin
goarch: arm64
pkg: github.com/klaytn/klaytn/blockchain/vm
BenchmarkOpMcopy-10    	     200	       115.6 ns/op
PASS
ok  	github.com/klaytn/klaytn/blockchain/vm	0.427s
```

[the remark this answers is not in the document]
Comment (yoomee1313) [reply blockchain/vm/eips.go:226]: I think callDataCopy computation cost seems a little inaccurate. I think you can configure the value by doubling the benchmark value.

File: blockchain/vm/instructions_test.go [hunk 3]
```diff
@@ -1398,3 +1400,144 @@ func fillStacks(stacks []string, n int) []string {
 	}
 	return stacks
 }
+func BenchmarkOpMcopy(b *testing.B) {
```
Comment (hyeonLewis) [inline blockchain/vm/instructions_test.go:1403]:
```suggestion

func BenchmarkOpMcopy(b *testing.B) {
```

**Review by blukat29 — APPROVED** [review]

File: blockchain/vm/eips.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (sjnam) [reply blockchain/vm/eips.go:226]: set to `250`

**Review by ian0371 — APPROVED** [review]

**Review by yoomee1313 — APPROVED** [review]

**Review by hyeonLewis — APPROVED** [review]

**Pull request merged**